### ISSUE TITLE
macOS installer tests

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nix#1739]
Another possibly less obvious benefit to merging @zimbatm's Travis PR is that we could finally start testing the macOS installer exhaustively. As far as I can tell right now, in the Travis config we can currently request macOS 10.10 through 10.12 (presumably 10.13 is coming soon), and a single build can trigger on multiple platforms, so we could conceivably test that the macOS installer works on all of them. They all run in isolated throwaway VMs so we can make systemwide changes (right?) and generally test that the installers work properly.

tl;dr: even if we completely disregard the PR testing benefits of Travis, I think the whole "throwaway macOS machines" thing makes it a slam dunk, especially with the macOS installer pain we've been experiencing recently.

cc @edolstra @grahamc @LnL7 @shlevy @domenkozar

**Comment by grahamc** [comment]
Note there _already is_ a .travis.yml which I use for testing the installer: https://github.com/NixOS/nix/blob/master/.travis.yml anyone can use this without it being enabled on the main repo, by enabling Travis on your fork.

**Comment by domenkozar** [comment]
I use something like https://github.com/domenkozar/test-nix-1-11-7/blob/master/.travis.yml

**Comment by copumpkin** [comment]
Oh, very cool, I didn't realize (or perhaps forgot?) that it was already there.

**Comment by domenkozar** [comment]
I've implemented this in https://github.com/NixOS/nix/pull/4549
